### PR TITLE
Integrate QuicTransport server with other wpt tools

### PR DESCRIPTION
**Pull request by Hexcles**
Now there are four ways to start the QUIC server, each one delegating to the previous layer:

1. (Standalone mode) Run `python3 tools/quic/serve.py` directly. This isn't really useful as you'd need to install the dependencies yourself.
2. `wpt serve-quic-transport` brings up only the QUIC server, but using the `wpt` frontend to manage virtualenv automatically (`_venv3`). This is mostly for internal plumbing only.
3. `wpt serve --quic-transport` brings up `wptserve` as well as `wpt serve-quic-transport`. This is the main entry point. `--quic-transport` won't be needed once the QUIC server is enabled by default.
4. `wpt run --enable-quic ...` can run tests that require QUIC servers (currently just the QUIC Transport server) and essentially uses `wpt serve --quic-transport` under the hood.

### CODE REVIEW

**Comment by Hexcles** [comment]
@yutakahirano I've got almost everything up and running now, except this error from aioquic:

```
INFO:root:QUIC event: <class 'aioquic.quic.events.ProtocolNegotiated'>                                                                                                                                                                                                                                                   
INFO:quic:[fd2aec89dd8db3dc] Connection close code 0xA, reason 28:TLS handshake failure (ENCRYPTION_HANDSHAKE) 46: certificate unknown                                                                                                                                                         
INFO:root:QUIC event: <class 'aioquic.quic.events.ConnectionTerminated'>
```

(I'm using `-c tools/quic/certs/cert.pem -k tools/quic/certs/cert.key`.)

**Comment by yutakahirano** [comment]
Sorry for the late reply.

@Hexcles did you give appropriate commandline parameters to Chromium? [1] maybe helpful. Also, recent Chrome canary prints an error message to the console log on handshake error. Can you tell me what it says?

1: https://docs.google.com/document/d/1vnv4kwiGesCAaAjFlXR1XZHN3YeHe9xZcTmIvSoX1aU/edit#heading=h.ru2qqp8h0kx2

**Comment by yutakahirano** [comment]
I have one question. Which do you prefer, "quic" or "quic-transport"? QuicTransport(https://tools.ietf.org/html/draft-vvv-webtransport-quic) is a protocol built on top of QUIC so QUIC may be misleading.

My original intention was having QUIC related tools (including QuicTransport related ones) under tools/quic. webtransport/quic is not a problem because "QUIC" in the context of WebTransport surely means QuicTransport.

But I'm not sure unqualified "QUIC" (e.g., wpt serve-quic) is good.

File: tools/quic/quic_transport_server.py [hunk 1]
```diff
@@ -167,73 +167,29 @@ def pop(self, label: bytes) -> Optional[SessionTicket]:
...
     configuration = QuicConfiguration(
         alpn_protocols=['wq-vvv-01'] + ['siduck'],
         is_client=False,
         max_datagram_frame_size=65536,
     )
 
-    handlers_path = os.path.abspath(os.path.expanduser(args.handlers_path))
-    logging.log(logging.INFO, 'port = %s' % args.port)
-    logging.log(logging.INFO, 'handlers path = %s' % handlers_path)
+    global handlers_path
```
Comment (Hexcles) [inline tools/quic/quic_transport_server.py:177]: Missing this was a bug that caused "Client indication error: unsupported operand type(s) for +: 'NoneType' and 'str'".

File: tools/serve/serve.py [hunk 2]
```diff
@@ -702,6 +704,63 @@ def start_wss_server(host, port, paths, routes, bind_address, config, **kwargs):
...
+def start_quic_transport_server(host, port, paths, routes, bind_address, config, **kwargs):
+    # Ensure that when we start this in a new process we have the global lock
+    # in the logging module unlocked
+    reload_module(logging)
+    release_mozlog_lock()
+    try:
+        return QuicTransportDaemon(host,
+                          port,
+                          private_key=config.ssl_config["key_path"],
+                          certificate=config.ssl_config["cert_path"],
```
Comment (Hexcles) [inline tools/serve/serve.py:758]: Note that this means when running `wpt serve --quic-transport` we use the certs from `tools/certs` by default, which now works with aioquic 0.8.8.

**Review by Hexcles — COMMENTED** [review]
@yutakahirano Thanks for the link. I didn't know "--ignore-certificate-errors does not work for this feature". Explicitly specifying SPKI works.

Now with the right flags, Chrome 84 Dev passes `webtransport/quic/client-indication.sub.any.html`. Server was started with `./wpt serve --quic-transport`.

PTAL again. I've highlighted a few critical changes.

File: tools/quic/quic_transport_server.py [hunk 3]
```diff
@@ -16,8 +15,9 @@
 from aioquic.tls import SessionTicket
 
 SERVER_NAME = 'aioquic-transport'
+logger = logging.getLogger(__name__)
 
-handlers_path = None
+handlers_path = ""
```
Comment (yutakahirano) [inline tools/quic/quic_transport_server.py:20]: Single quotations are used in this file so doing so here is probably better for consistency.

File: tools/quic/quic_transport_server.py (hunk 1 above)
Comment (yutakahirano) [reply tools/quic/quic_transport_server.py:177]: Yeah, previously this code was in the global scope but now it's in a function.

**Review by yutakahirano — APPROVED** [review]
Thank you, LGTM!

File: tools/quic/quic_transport_server.py (hunk 3 above)
Comment (Hexcles) [reply tools/quic/quic_transport_server.py:20]:
```suggestion
handlers_path = ''
```

**Pull request merged**